manifests/vhost.pp

```diff
@@ -263,8 +267,12 @@
 
   # Same as above, but for logroot
   if ! defined(File[$logroot]) {
+    $logroot_ensure_real = $ensure? {
+      present => $logroot_ensure,
+      default => 'absent',
+    }
     file { $logroot:
-      ensure  => $logroot_ensure,
+      ensure  => $logroot_ensure_real,
```

Comment:
same thing here as the ```docroot_ensure``` comment. defaulting to ```$ensure``` instead of the current ```'directory'``` might correct it in a smooth, convenient way.